models/MARBL_joint_estimation/model_mod.f90

```diff
@@ -0,0 +1,654 @@
+    table(i,5) = trim(update)
+
+    if ( table(i,1) == ' ' .and. table(i,2) == ' ' .and. table(i,3) == ' ' .and. table(i,4) == ' ' .and. table(i,5) == ' ') exit MyLoop ! Found end of list.
+
+    if ( table(i,1) == ' ' .or. table(i,2) == ' ' .or. table(i,3) == ' ' .or. table(i,4) == ' ' .or. table(i,5) == ' ') then
+        string1 = 'model_nml:model_state_variables not fully specified'
+        call error_handler(E_ERR,'verify_state_variables',string1)
+    endif
+
+    ! Make sure DART qty is valid
+
+    qty_list(i) = get_index_for_quantity(dartstr)
+    if( qty_list(i)  < 0 ) then
+        write(string1,'(''there is no obs_kind <'',a,''> in obs_kind_mod.f90'')') trim(dartstr)
+        call error_handler(E_ERR,'verify_state_variables',string1)
+    endif
+    
+    ! Make sure the update variable has a valid name
+
+    select case (update)
+        case ('UPDATE')
+            update_var(i) = .true.
+        case ('NO_COPY_BACK')
+            update_var(i) = .false.
+        case default
+            write(string1,'(A)')  'only UPDATE or NO_COPY_BACK supported in model_state_variable namelist'
+            write(string2,'(6A)') 'you provided : ', trim(varname), ', ', trim(dartstr), ', ', trim(update)
+            call error_handler(E_ERR,'verify_state_variables',string1, text2=string2)
+    end select
+
+    ! reading the clamp values
+
+    if (table(i, 3) /= 'NA') then
+        read(table(i,3), '(d16.8)') clamp_vals(i,1)
+    else
+        clamp_vals(i,1) = MISSING_R8
+    endif
+    
+    if (table(i,4) /= 'NA') then
+        read(table(i,4), '(d16.8)') clamp_vals(i,2)
+    else
+        clamp_vals(i,2) = MISSING_R8
+    endif
+
+    ngood = ngood + 1
+enddo MyLoop
+
+
+end subroutine verify_state_variables
+
+!------------------------------------------------------------
+function on_v_grid(qty)
+
+integer, intent(in)  :: qty
+logical :: on_v_grid
+
+if (qty == QTY_V_CURRENT_COMPONENT) then
+    on_v_grid = .true.
+else
+    on_v_grid = .false.
+endif
+
+end function on_v_grid
+
+!----------------------------------------------------------
+function on_u_grid(qty)
+
+integer, intent(in)  :: qty
+logical :: on_u_grid
+
+if (qty == QTY_U_CURRENT_COMPONENT) then
+    on_u_grid = .true.
+else
+    on_u_grid = .false.
+endif
+
+end function on_u_grid
+
+!------------------------------------------------------------
+! Read number of vertical layers from mom6 template file
+subroutine read_num_layers()
+
+integer :: ncid
+
+character(len=*), parameter :: routine = 'read_num_layers'
+
+ncid = nc_open_file_readonly(template_file(1))
+
+call nc_get_variable_size(ncid, 'Layer', nz)
+
+call nc_close_file(ncid)
+
+end subroutine read_num_layers
+
+!------------------------------------------------------------
+! ocean_geom are 2D state sized static data
+! HK Do these arrays become too big in high res cases?
+subroutine read_ocean_geometry()
+
+integer :: ncid
+
+character(len=*), parameter :: routine = 'read_ocean_geometry'
+
+! Need nx, ny
```

Comment:
```suggestion
```
I think this comment is from the MOM6 model_mod where you need nx, ny for the size of the basin_depth variable. The column marble is (2,2) always.